runtime/syntax/python.vim

```diff
@@ -157,17 +157,17 @@ syn match   pythonEscape	"\\$"
 " and so on, as specified in the 'Python Language Reference'.
 " https://docs.python.org/reference/lexical_analysis.html#numeric-literals
 if !exists("python_no_number_highlight")
-  " numbers (including longs and complex)
-  syn match   pythonNumber	"\<0[oO]\=\o\+[Ll]\=\>"
-  syn match   pythonNumber	"\<0[xX]\x\+[Ll]\=\>"
-  syn match   pythonNumber	"\<0[bB][01]\+[Ll]\=\>"
-  syn match   pythonNumber	"\<\%([1-9]\d*\|0\)[Ll]\=\>"
-  syn match   pythonNumber	"\<\d\+[jJ]\>"
-  syn match   pythonNumber	"\<\d\+[eE][+-]\=\d\+[jJ]\=\>"
+  " numbers (including complex)
+  syn match   pythonNumber	"\<0[oO]\%(_\=\o\)\+\>"
```

Comment:
Is it intentional to only support only python3 numbers and not python2?

For example, these python2 numbers are not highlighted correctly in this PR:
```
0123  # octal, in Python3 is should 0o123
123l  # long, in Python3 is should be 123
123L  # long, in Python3 it should be 123
```
My original PR https://github.com/vim/vim/pull/8024 supported the union of python2 and python3 numbers.
I realize that python2 is end if life, but there are still plenty of python2 scripts.
Maybe we need something like `if exists("python2_highlight")` for python2 specific things?

Reply:
@dpelle Yes, this **is intentional**. I already mentioned that in the review of your PR. 🙂 

I _already_ dropped the support for Python 2 _keywords_ and _builtins_ in PR #7855.
My intent, as stated in that PR is to continue updating the syntax highlighting with the Python 3 features that are missing right now, while at the same time _dropping_ the Python 2 features. This PR is one of the steps in that process that has already been started with PR #7855.

**Rationale**: By showing the users that their code is using Python 2 syntax through the lack of syntax highlighting for those keywords/builtins or `123L` numbers, or in the next PR I'm preparing of `ur"some string"`, we will encourage the users to transition to Python 3. As a long-time member of Python community I see that as my obligation.

*Anyone* who needs to use Python 2 can take the copy of the syntax file *before* PR #7855 and use it locally.
Python 2 is _unsupported_ for over a year now, and I intend to support only the official language features -- Python 3.
The examples you provided are syntactically incorrect.
```
>>> 123L
  File "<stdin>", line 1
    123L
       ^
SyntaxError: invalid syntax
```
Yes, I know that it would work with Python 2 interpreter, but I want to discourage everyone from using it.
Please, do notice that this is not just my position, but the position of the Python Foundation and Python Steering Council.

Reply:
To demonstrate the intent, below is the part of my tests relevant to this change.
Notice the last section of the _bad_ examples, related to Python 2 features that are now syntactically incorrect.
```
good = (
    # From Python docs
    7, 2147483647, 0o177, 0b100110111,
    3, 79228162514264337593543950336, 0o377, 0xdeadbeef,
    100_000_000_000, 0b_1110_0101,
    3.14, 10., .001, 1e100, 3.14e-10, 0e0, 3.14_15_93,
    3.14j, 10.j, 10j, .001j, 1e100j, 3.14e-10j, 3.14_15_93j,

    # From PEP 515
    10_000_000.0, 0xCAFE_F00D, 0b_0011_1111_0100_1110, 0b_1111_0000,

    # my tests
    23, 45678, 0, 1_000_000_000_000_000, 0_0_0_0_0_0,
    0xf, 0xab123, 0XABCDEF, 0xDEADBEEF, 0x_f, 0x_FF_FF_FF_FF,
    0o17, 0o077, 0o377, 0O245, 0o456, 0o_75, 0O177_377,
    0b101, 0B1010, 0b1111111111111, 0b_1101,
    2.23, 1., 1.0, .1, 088e-08, 231_782.32_26, 231_782.32_26e3_0,
    1e3, 1.e+5, 1.0e-2, 2e0, 0e0, 1e3_2, 1e-3_2, .2_6E-1_3, 1.2_6e1_3,
    08j, 1j, 1.j, 1.0j, 1e3j, 1.e+5J, 1.0e-2j, 3.14+42.42j, 1_2.5_6e3_4j,
    -1, +3, -1.45E13J, 4.,
)

bad = (
    1.0.0, 08, 0xg, 0o8, 0b2,
    a=-1, b=+2, c=4,
    $1, j1, A.1, 1.0a, 2009.02.15,

    23_, _23, 2__3, _0, 0_, 0__0_0,
    0xf_, _0xf, 0_xf, 0x__f, 0xff__ff,
    0o7_, _0o7, 0_o7, 0o__7, 0o7__7,
    0b1_, _0b1, 0_b1, 0b__1, 0b1__1,
    1_000_000_, 1_000__000, 0o177__377, 0b_1110_,
    231__782.32, 2_.3, _2.3, 231_782.32__26, 2._3, 2.3_,
    1e_3, 1_e3, 1e3_, _1.2e3, j, _1j, _1.2j,

    # Python 2
    23L, 45678l,
    0xfL, 0XDEADBEEFL,
    0177, 0377, 0245L, 0o456L,
    0b1111111111111L,
    +4L,
)
```